### PR TITLE
add getUserFavoriteGames

### PR DESCRIPTION
**Pull request by AbdullahalyDev**

### CODE REVIEW

File: lib/accountinformation/getUserFavoriteGames.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const http = require('../util/http.js').func
+
+// Args
+exports.required = ['userId']
+exports.optional = ['accessFilter', 'limit', 'cursor', 'sortOrder']
+
+// Docs
+/**
+ * ✅ Get the favorite games data.
+ * @category AccountInformation
```
Comment (suufi) [inline lib/accountinformation/getUserFavoriteGames.js:11]: Not the right category

File: lib/accountinformation/getUserFavoriteGames.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @alias getUserFavoriteGames
+ * @param {number} userId - The id of the user.
+ * @param {number} limit - The number of results per request.
+ * @returns {Promise<UserFavoriteGames>}
+ * @example
+ * const noblox = require("noblox.js")
+ * const userSocialLinks = await noblox.getUserFavoriteGames(172694510);
+**/
+
+function getUserFavoriteGames (userId, limit) {
```
Comment (suufi) [inline lib/accountinformation/getUserFavoriteGames.js:21]: missing multiple parameters specified by the endpoint

see: https://games.roblox.com/docs/index.html

File: lib/accountinformation/getUserFavoriteGames.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      }
+      throw new Error(`Errors Found:\n  ${errors.join('\n  ')}\n`)
+    } else {
+      return body
+    }
+  })
+}
+
+exports.func = function ({ userId, accessFilter, limit, cursor, sortOrder }) {
+  return getUserFavoriteGames(userId, accessFilter, limit, cursor, sortOrder)
```
Comment (suufi) [inline lib/accountinformation/getUserFavoriteGames.js:42]: should be handled with getPageResults

File: test/accountinformation.test.js [hunk 4]
```diff
@@ -9,6 +9,22 @@ beforeAll(() => {
 })
 
 describe('Account Information Methods', () => {
+
+  it ('getUserFavoriteGame() return a player\' favorite games', () => {
+    return getUserFavoriteGames(172694510).then((res) => {
+      return expect(res).toMatchObject({
+        previousPageCursor: expect.toBeOneOf([expect.any(String), null]),
+        nextPageCursor: expect.toBeOneOf([expect.any(String), null]),
+        data: expect.toBeOneOf([expect.any(Object), null]),
```
Comment (suufi) [inline test/accountinformation.test.js:18]: needs to be more specific than this

File: test/accountinformation.test.js [hunk 5]
```diff
@@ -9,6 +9,22 @@ beforeAll(() => {
 })
 
 describe('Account Information Methods', () => {
+
+  it ('getUserFavoriteGame() return a player\' favorite games', () => {
```
Comment (suufi) [inline test/accountinformation.test.js:13]: player's*

File: test/accountinformation.test.js [hunk 6]
```diff
@@ -9,6 +9,22 @@ beforeAll(() => {
...
+      return expect(res).toMatchObject({
+        previousPageCursor: expect.toBeOneOf([expect.any(String), null]),
+        nextPageCursor: expect.toBeOneOf([expect.any(String), null]),
+        data: expect.toBeOneOf([expect.any(Object), null]),
+      })
+    })
+  })
+
+  it('getUserFavoriteGame() doesn\'t return a player\'s favorite games and errors when user is invalid', async () => {
+    return await expect(getUserFavoriteGames()).rejects.toThrow()
```
Comment (suufi) [inline test/accountinformation.test.js:24]: you should actually pass an invalid user id here

File: typings/index.d.ts [hunk 7]
```diff
@@ -1375,6 +1392,11 @@ declare module "noblox.js" {
 
     /// AccountInformation
 
+    /**
```
Comment (suufi) [inline typings/index.d.ts:1395]: not the right category, should be under games

File: typings/index.d.ts [hunk 8]
```diff
@@ -215,6 +215,23 @@ declare module "noblox.js" {
         CollectibleItemId?: number;
     }
 
+    type UserFavoriteGames = {
```
Comment (suufi) [inline typings/index.d.ts:218]: interface*

**Review by suufi — CHANGES_REQUESTED** [review]

File: lib/games/getUserFavoriteGames.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+// Docs
+/**
+ * ✅ Get the favorite games data.
+ * @category Games
+ * @alias getUserFavoriteGames
+ * @param {number} userId - The id of the user.
+ * @param {number} accessFilter - The  number of results per request.
+ * @param {number} limit - The  number of results per request.
+ * @param {number} cursor - The  number of results per request.
```
Comment (Neztore) [inline lib/games/getUserFavoriteGames.js:16]: Cursor parameter description

File: lib/games/getUserFavoriteGames.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+exports.required = ['userId']
+exports.optional = ['accessFilter', 'limit', 'cursor', 'sortOrder']
+
+// Docs
+/**
+ * ✅ Get the favorite games data.
+ * @category Games
+ * @alias getUserFavoriteGames
+ * @param {number} userId - The id of the user.
+ * @param {number} accessFilter - The  number of results per request.
```
Comment (Neztore) [inline lib/games/getUserFavoriteGames.js:14]: Access filter description

File: lib/games/getUserFavoriteGames.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+ * @param {number} limit - The  number of results per request.
+ * @param {number} cursor - The  number of results per request.
+ * @param {number} sortOrder - The  number of results per request.
```
Comment (Neztore) [inline lib/games/getUserFavoriteGames.js:17]: sortOrder description

**Review by Neztore — CHANGES_REQUESTED** [review]
Looks good! A few things to fix with the docs though.

**Comment by AbdullahalyDev** [comment]
.

**Pull request closed without merging**